lib/rubocop/cop/lint/shadowing_outer_local_variable.rb

```diff
@@ -57,10 +57,20 @@ def before_declaring_variable(variable, variable_table)
 
           outer_local_variable = variable_table.find_variable(variable.name)
           return unless outer_local_variable
+          return if same_conditions_node_different_branch?(variable, outer_local_variable)
 
           message = format(MSG, variable: variable.name)
           add_offense(variable.declaration_node, message: message)
         end
+
+        def same_conditions_node_different_branch?(variable, outer_local_variable)
+          variable_node = variable.scope.node.parent
+          outer_local_variable_node = outer_local_variable.scope.node
+
+          variable_node.conditional? &&
+            outer_local_variable_node.conditional? &&
+            variable_node == outer_local_variable_node
```

Comment:
The following reduces the processing that may be unnecessary.

```suggestion
          variable_node = variable.scope.node.parent
          return false unless variable_node.conditional?

          outer_local_variable_node = outer_local_variable.scope.node

          outer_local_variable_node.conditional? && variable_node == outer_local_variable_node
```

Reply:
I see it better that way too!
Update PR. Thank you so much.